commit c0a881204f3a
Author: Martin Stransky <stransky@redhat.com>
Date:   Thu May 17 14:40:21 2018 +0200

    Bug 1461203 - Use GetScrollbarMetrics() to access scrollbar metrics. r=jhorak, a=RyanVM
    
    MozReview-Commit-ID: 3gnFNfxQJZi
    
    --HG--
    extra : source : bdfce06ecdb44275214caa0c8333d3112bbb4d16
---
 widget/gtk/gtk3drawing.cpp | 14 ++++++--------
 1 file changed, 6 insertions(+), 8 deletions(-)

diff --git widget/gtk/gtk3drawing.cpp widget/gtk/gtk3drawing.cpp
index 8efc4f70d1be2..71ddd1e9d4bb7 100644
--- widget/gtk/gtk3drawing.cpp
+++ widget/gtk/gtk3drawing.cpp
@@ -888,16 +888,14 @@ moz_gtk_scrollbar_button_paint(cairo_t *cr, const GdkRectangle* aRect,
         // in case of horizontal scrollbars).
         GtkOrientation orientation = flags & MOZ_GTK_STEPPER_VERTICAL ?
             GTK_ORIENTATION_VERTICAL : GTK_ORIENTATION_HORIZONTAL;
-        const auto& metrics = sScrollbarMetrics[orientation];
-        if (!metrics.initialized) {
-            NS_WARNING("Didn't measure before drawing?");
-        }
+
+        const ScrollbarGTKMetrics* metrics = GetScrollbarMetrics(orientation);
         if (flags & MOZ_GTK_STEPPER_VERTICAL) {
-            rect.x += metrics.border.track.left;
-            rect.width = metrics.size.thumb.width;
+            rect.x += metrics->border.track.left;
+            rect.width = metrics->size.thumb.width;
         } else {
-            rect.y += metrics.border.track.top;
-            rect.height = metrics.size.thumb.height;
+            rect.y += metrics->border.track.top;
+            rect.height = metrics->size.thumb.height;
         }
     }
 
